advanced/Scripts/chronometer.sh

```diff
@@ -8,98 +8,428 @@
+      *) sys_model="" ;;
+    esac
+    sys_type="Raspberry Pi$sys_model"
+  else
+    source "/etc/os-release"
+    CODENAME=$(sed 's/[()]//g' <<< "${VERSION/* /}")
+    sys_type="${NAME/ */} ${CODENAME^} $VERSION_ID"
+  fi
+
+  # Get core count
+  sys_cores=$(grep -c "^processor" /proc/cpuinfo)
+  [[ "$sys_cores" -ne 1 ]] && sys_cores_plu="cores" || sys_cores_plu="core"
+
+  # Test existence of clock speed file for ARM CPU
+  if [[ -f "/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq" ]]; then
+    scaling_freq_file="/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq"
+  fi
+
+  # Test existence of temperature file
+  if [[ -f "/sys/class/thermal/thermal_zone0/temp" ]]; then
+    temp_file="/sys/class/thermal/thermal_zone0/temp"
+  elif [[ -f "/sys/class/hwmon/hwmon0/temp1_input" ]]; then 
+    temp_file="/sys/class/hwmon/hwmon0/temp1_input"
+  else
+    temp_file=""
+  fi
+
+  # Test existence of setupVars config
+  if [[ -f "/etc/pihole/setupVars.conf" ]]; then
+    setupVars="/etc/pihole/setupVars.conf"
+  fi
 }
 
-normalChrono() {
+get_sys_stats() {
+  local ph_ver_raw
+  local cpu_raw
+  local ram_raw
+  local disk_raw
+
+  # Update every 12 refreshes (Def: every 60s)
+  count=$((count+1))
+  if [[ "$count" == "1" ]] || (( "$count" % 12 == 0 )); then
+    [[ -n "$setupVars" ]] && source "$setupVars"
+    
+    
+    ph_ver_raw=($(pihole -v -c 2> /dev/null | sed -n 's/^.* v/v/p'))
+    if [[ -n "${ph_ver_raw[0]}" ]]; then
+      ph_core_ver="${ph_ver_raw[0]}"
+      ph_lte_ver="${ph_ver_raw[1]}"
+      ph_ftl_ver="${ph_ver_raw[2]}"
+    else
+      ph_core_ver="${COL_LIGHT_RED}API unavailable${COL_NC}"
+    fi
+    
+    sys_name=$(hostname)
+    
+    [[ -n "$TEMPERATUREUNIT" ]] && temp_unit="$TEMPERATUREUNIT" || temp_unit="c"
+    
+    # Get storage stats for partition mounted on /
+    disk_raw=($(df -B1 / 2> /dev/null | awk 'END{ print $3,$2,$5 }'))
+    disk_used="${disk_raw[0]}"
+    disk_total="${disk_raw[1]}"
+    disk_perc="${disk_raw[2]}"
+    
+    net_gateway=$(route -n | awk '$4 == "UG" {print $2;exit}')
+    
+    # Get DHCP stats, if feature is enabled
+    if [[ "$DHCP_ACTIVE" == "true" ]]; then
+      ph_dhcp_eip="${DHCP_END##*.}"
+      ph_dhcp_max=$(( ${DHCP_END##*.} - ${DHCP_START##*.} + 1 ))
+    fi
+    
+    # Get alt DNS server, or print total count of alt DNS servers
+    if [[ -z "${PIHOLE_DNS_3}" ]]; then
+      ph_alts="${PIHOLE_DNS_2}"
+    else
+      dns_count="0"
+      [[ -n "${PIHOLE_DNS_2}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_3}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_4}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_5}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_6}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_7}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_8}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_9}" ]] && dns_count="$dns_count+"
```

Comment:
I wasn't totally sure what the limit of upstream DNS servers was, so I made it show a maximum of "9+"